### ISSUE TITLE
CaseClassDiffs._ import only influences the top level of the structure in Scala 2.12

### ISSUE THREAD

**Issue by nigredo-tori** [etorreborre/specs2#900]
[Scastie example](https://scastie.scala-lang.org/lL88X1GaSGKeVwBn2B1JTQ).

Consider this code:
```scala
import org.specs2.matcher.describe.Diffable
import org.specs2.matcher.CaseClassDiffs._

final case class Foo(x: Int, y: String)
final case class Bar(foo: Foo)

val bar1 = Bar(Foo(123, "abc"))
val bar2 = Bar(Foo(456, "def"))

println(Diffable.diff(bar1, bar2).render(""))
```
With Scala 2.12 this prints:
```
Bar(foo: Foo(123,abc) != Foo(456,def))
```
Note that while `Bar` uses a "nice" `Diffable`, `Foo` defaults to the single-line equality check!

There is a workaround. Adding this line:
```scala
implicit val FooDiffable: Diffable[Foo] = caseClassDiffable
```
fixes the output:
```
Bar(foo: Foo(x: 123 != 456,
             y: 'abc' != 'def'))
```

The issue only arises with Scala 2.12.

**Comment by etorreborre** [comment]
Thanks for the report. Implicit resolution with those implicits can be a bit tricky. If it works fine with 2.13, I don't know what we can do except leaving this here as a workaround. On the other hand I have been reworking the case class diff for Scala 3 (since there's a better support for metaprogramming) and I will make sure that this case still works

**Comment by etorreborre** [comment]
This is fixed on the latest `specs2-5.0.0-RC-03`
